File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var maxVowels = function(s, k) {
    var vowel_set = new Set(['a', 'e', 'i', 'o', 'u']);
    let curr_size = 0, max_cnt = 0;
    for (var i = 0; i < k; i++) {
        if (vowel_set.has(s[i])) {
            curr_size++;
        }
    }

    max_cnt = curr_size;

    for (var i = k; i < s.length; i++) {
        if (vowel_set.has(s[i - k])) {
            curr_size--;
        }
        if (vowel_set.has(s[i])) {
            curr_size++;
        }

        max_cnt = Math.max(max_cnt, curr_size);
    }

    return max_cnt;
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {

        int max_count = 0;
        int curr_count = 0;
        int i = 0;
        HashSet<Character> vowel = new HashSet<Character>();

        vowel.add('a');
        vowel.add('e');
        vowel.add('i');
        vowel.add('o');
        vowel.add('u');

        for (i = 0; i < k; i++) {
            if (vowel.contains(s.charAt(i))) {
                curr_count++;
            }
        }

        max_count = curr_count;

        while (i >= k && i < s.length()) {
            if (vowel.contains(s.charAt(i-k))) {
                curr_count--;
            }
            if (vowel.contains(s.charAt(i))) {
                curr_count++;
            }
            max_count = Math.max(max_count, curr_count);
            i++;
        }

        return max_count;
    }
}